### PR TITLE
Add automation tool for generating synthetic events data

### PR DESCRIPTION
**Pull request by natekupp**

### CODE REVIEW

File: python_modules/automation/generate_synthetic_events.py [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+                    'session': secrets.token_urlsafe(16),
+                    'persistent': secrets.token_urlsafe(16),
+                },
+                'run_id': self.faker.uuid4(),
+                'type': self._gen_event_type(),
+                'user_agent': self._gen_user_agent(),
+                'ip_address': self.faker.ipv4_public(),
+                'timestamp': self._gen_timestamp(),
+                'url': '/' + self.faker.uri_path(),
+                # like any good production system, we throw some random PII in our behavioral events
```
Comment (mgasner) [inline python_modules/automation/generate_synthetic_events.py]: yessss

File: python_modules/automation/generate_synthetic_events.py [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+    required=True,
+)
+@click.option('--num-files', default=100, help='Total number of event files.', required=True)
+@click.option(
+    '--num-events-per-file',
+    default=100000,
+    help='Batch size, number of events per file.',
+    required=True,
+)
+@click.option('--dry-run', is_flag=True, help='Dry runs won\'t write to S3.')
```
Comment (mgasner) [inline python_modules/automation/generate_synthetic_events.py]: dry runs create the files and then delete them? why bother?

File: python_modules/automation/generate_synthetic_events.py [hunk 3]
```diff
@@ -0,0 +1,160 @@
...
+def run(s3_bucket, s3_prefix, start_date, end_date, num_files, num_events_per_file, dry_run):
+    start = datetime.datetime.strptime(start_date, "%Y-%m-%d")
+    end = datetime.datetime.strptime(end_date, "%Y-%m-%d")
+    dates = [start + datetime.timedelta(days=x) for x in range(0, (end - start).days)]
+
+    # Synthesize date files first
+    with Pool(processes=NUM_PROCESSES) as p:
```
Comment (schrockn) [inline python_modules/automation/generate_synthetic_events.py]: this is cool

File: python_modules/automation/generate_synthetic_events.py (hunk 2 above)
Comment (natekupp) [reply python_modules/automation/generate_synthetic_events.py]: ah good catch, I meant to put the os.unlink() under the `if not dry_run`, will update - meant to permit inspecting the files

**Pull request merged**